File: src/main/java/de/maggiwuerze/xdccloader/service/UserSettingsService.java
package de.maggiwuerze.xdccloader.service;

import de.maggiwuerze.xdccloader.model.entity.UserSettings;
import de.maggiwuerze.xdccloader.persistence.IrcUserRepository;
import de.maggiwuerze.xdccloader.persistence.SettingsRepository;
import de.maggiwuerze.xdccloader.persistence.UserSettingsRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
@Slf4j
public class UserSettingsService {

	private final SettingsRepository settingsRepository;
	private final IrcUserRepository ircUserRepository;
	private final UserSettingsRepository userSettingsRepository;

	public UserSettings getUserSettings() {
		return settingsRepository.getFirstById(1L);
	}

	public void saveUserSettings(UserSettings userSettingsBy) {
		settingsRepository.save(userSettingsBy);
	}

	public void setInitialized(boolean b) {
		UserSettings userSettings = getUserSettings();
		userSettings.setInitialized(true);
		saveUserSettings(userSettings);
	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/service/UserSettingsService.kt
package de.maggiwuerze.xdccwebloader.service

import de.maggiwuerze.xdccwebloader.persistence.entity.UserSettings
import de.maggiwuerze.xdccwebloader.persistence.IrcUserRepository
import de.maggiwuerze.xdccwebloader.persistence.UserSettingsRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserSettingsService
    (
    private val ircUserRepository: IrcUserRepository,
    private val settingsRepository: UserSettingsRepository,
) {

    fun getUserSettings(usersettingsId: UUID? = null): UserSettings =
        usersettingsId?.let {
            settingsRepository.getFirstById(usersettingsId)
        } ?: settingsRepository.findAll().firstOrNull() ?: throw IllegalStateException("UserSettings not found!")

    fun saveUserSettings(userSettingsBy: UserSettings) {
        settingsRepository.save(userSettingsBy)
    }

    fun setInitialized() {
        val usersettings = getUserSettings()
        usersettings.initialized = true
        saveUserSettings(usersettings)
    }
}
